nucliadb_node/src/replication/replicator.rs

```diff
@@ -146,7 +146,7 @@ pub async fn replicate_shard(
     // We only do this here because GC will not be done otherwise on a secondary
     // XXX this should be removed once we refactor gc/merging
     let sshard = Arc::clone(&shard); // moved shard for gc
-    tokio::task::spawn_blocking(move || sshard.gc()).await?.expect("GC failed");
+    tokio::task::spawn_blocking(move || sshard.collect_garbage()).await?.expect("GC failed");
```

Comment:
This will GC. A writer is running so GC will run. However, the writer has not reloaded the state so the GC will delete everything new.